src/go/k8s/cmd/configurator/main.go

```diff
@@ -109,11 +108,15 @@ func main() {
 
 	log.Print(c.String())
 
-	fs := afero.NewOsFs()
-	p := config.Params{ConfigPath: path.Join(c.configSourceDir, "redpanda.yaml")}
-	cfg, err := p.Load(fs)
+	p := path.Join(c.configSourceDir, "redpanda.yaml")
+	cf, err := os.ReadFile(p)
 	if err != nil {
-		log.Fatalf("%s", fmt.Errorf("unable to read the redpanda configuration file: %w", err))
+		log.Fatalf("%s", fmt.Errorf("unable to read the redpanda configuration file, %q: %w", p, err))
+	}
+	cfg := &config.Config{}
+	err = yaml.Unmarshal(cf, cfg)
+	if err != nil {
+		log.Fatalf("%s", fmt.Errorf("unable to parse the redpanda configuration file, %q: %w", p, err))
```

Comment:
This is from #7107 as well (and the go.mod change)

Reply:
yes, sorry, the rebase is a bit messed up. fixed in the last push